comms/EclMeta.cpp

```diff
@@ -0,0 +1,309 @@
+        {
+            std::_tstring pather = meta->GetPath().c_str();
+            if (boost::algorithm::equals(pather, path))
+            {
+                return (CEclFile *)meta;
+            }
+        }
+    }
+    return NULL;
+}
+
+void CEclMeta::Update(const std::wstring & xml)
+{
+    CComInitialize com;
+    CComPtr<ISAXXMLReader> reader;
+    HRESULT hr = reader.CoCreateInstance(CLSID_SAXXMLReader30);
+
+    CComPtr <CECLCCMetaHandler> handler = new CECLCCMetaHandler(this);
+    hr = reader->putContentHandler(handler);
+    if (hr == S_OK)
+    {
+        reader->parse(CComVariant(xml.c_str()));
+    }
+}
+
+void CEclMeta::LoadMetaData(const WPathVector *folders)
+{
+    for (WPathVector::const_iterator itr = folders->begin(); itr != folders->end(); ++itr)
+    {
+        PopulateMeta(itr->first,_T(""));
+    }
+}
+
+void CEclMeta::PopulateMeta(const boost::filesystem::wpath & fileOrDir, const std::wstring dottedPath, int level)
+{
+    if (clib::filesystem::exists(fileOrDir))
+    {
+        std::wstring newDottedPath;
+        
+        if (level)
+        {
+            if (dottedPath.empty())
+                newDottedPath = fileOrDir.stem().c_str();
+            else
+                newDottedPath = dottedPath + _T(".") + fileOrDir.stem().c_str();
+        }
+
+        if (HasValidExtension(pathToWString(fileOrDir)))
+        {
+            m_masterMeta[newDottedPath] = new CEclFile(fileOrDir);
+        }
+        else if (boost::filesystem::is_directory(fileOrDir))
+        {
+            if (level)
+                m_masterMeta[newDottedPath] = new CEclFolder(fileOrDir);
+            boost::filesystem::directory_iterator end_itr;
+            for (boost::filesystem::directory_iterator itr(wpathToPath(fileOrDir)); itr != end_itr; ++itr)
+            {
+                PopulateMeta(pathToWString(*itr), newDottedPath, ++level);
+            }
+        }
+    }
+}
+
+int CEclMeta::NormalizeAutoC(IAttribute *attr)
+{
+    typedef boost::tokenizer<boost::char_separator<TCHAR>, std::_tstring::const_iterator, std::_tstring> tokenizer;
+    boost::char_separator<TCHAR> sep(_T("."));
+    tokenizer tokens(m_autoString, sep);
+    int tokenCount = 0;
+    CString tok;
+    std::_tstring autoString;
+    m_autoTokens.clear();
+
+    for (tokenizer::iterator tok_iter = tokens.begin(); tok_iter != tokens.end(); ++tok_iter)
+    {
+        tok = tok_iter->c_str();
+        if (tok.Compare(_T("$")) == 0)
+        {
+            tok = attr->GetModuleQualifiedLabel(true);
+        }
+        m_autoTokens.push_back(tok);
+        if (tokenCount)
+        {
+            autoString += _T(".");
+        }
+        autoString += tok;
+        tokenCount++;
+    }
+    m_autoLast = tok;
+    m_autoString = autoString;
+    FindImportAs();
+
+    return tokenCount;
+}
+
+void CEclMeta::BuildTokenStr()
+{
+    std::_tstring newTokenStr= _T("");
+    typedef  std::vector <CString >::iterator tIntIter;
+    int count = 0;
```

Comment:
Very Minor:  Might be better as `boolean first = true;`

Reply:
agreed